src/functions.php

```diff
@@ -734,6 +736,7 @@ function parallel(iterable $tasks): PromiseInterface
         $deferred->resolve($results);
     }
 
+    /** @var PromiseInterface<array<T>> Can't define `Deferred()` above, currently not supported by PHPStan, see https://github.com/phpstan/phpstan/issues/11032 */
     return $deferred->promise();
```

Comment:
I'd really like to define the `Deferred()` for the `parallel()` and `series()` functions, but this doesn't seem to be supported by PHPStan. I opened a ticket for this in PHPStan (see https://github.com/phpstan/phpstan/issues/11032), but I don't want to leave this PR open until this gets eventually fixed/improved in PHPStan.

@WyriHaximus noticed that this runs with PHPStan bleeding edge, but adding this to our configuration throws many more errors that we'd have to ignore in most cases. With this in mind, I don't think its a good idea to run this with bleeding edge as it's making it worse in my opinion.

My suggestion is to merge this as is and come back once this is supported in a new PHPStan version. This way the test suite is green again and we can file new pull requests.